common-service-core/src/main/java/com/ca/mfaas/util/UrlUtils.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package com.ca.mfaas.util;
+
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang.RandomStringUtils;
+
+import java.net.*;
+import java.util.function.Supplier;
+
+@Slf4j
+@UtilityClass
+public class UrlUtils {
+
+    private static final Supplier<String> messageSupplier = () -> "Invalid URL";
+
+
+    /**
+     * Remove slashes from input string parameter
+     *
+     * @param string input parameter
+     * @return input without removed trailing slashes.
+     */
+    public static String trimSlashes(String string) {
+        return string.replaceAll("^/|/$", "");
+    }
+
+    /**
+     * Substitute '\\W' with '-' in the input string and return the result
+     *
+     * @param url - input url to be encoded by the hard coded character substitution
+     * @return An url string with any non alpha-numeric characters substituted by '-'
+     */
+    public static String getEncodedUrl(String url) {
+        if (url != null) {
+            return url.replaceAll("\\W", "-");
+        } else {
+            return RandomStringUtils.randomAlphanumeric(10);
+        }
+    }
+
+
+    /**
+     * Checks validity of URL string.
+     * Invalid URL string will trigger throwing a InvalidParameterException encapsulating the message of originally thrown MalformedURLException
+     *
+     * @param url validated url
+     */
```

Comment:
use `@throws` for this method and the ones below

Reply:
Added